### ISSUE TITLE
ConcurrentModificationException inside getMessagesFromHistory

### ISSUE THREAD

**Issue by djkovrik** [webim/webim-android-sdk-demo#4]
After few pagination history requests app crashes with ConcurrentModificationException inside 
getMessagesFromHistory from MessageHolderImpl
```
java.util.ConcurrentModificationException
        at java.util.ArrayList$SubList.size(ArrayList.java:1057)
        at java.util.AbstractCollection.toArray(AbstractCollection.java:136)
        at java.util.Collections$UnmodifiableCollection.toArray(Collections.java:1075)
        at java.util.ArrayList.<init>(ArrayList.java:191)
        at com.webimapp.android.sdk.impl.MessageHolderImpl$2$1.receive(MessageHolderImpl.java:201)
        at com.webimapp.android.sdk.impl.MessageHolderImpl.respondMessages(MessageHolderImpl.java:238)
        at com.webimapp.android.sdk.impl.MessageHolderImpl.access$1100(MessageHolderImpl.java:23)
        at com.webimapp.android.sdk.impl.MessageHolderImpl$3.onSuccess(MessageHolderImpl.java:230)
        at com.webimapp.android.sdk.impl.WebimSessionImpl$RemoteHistoryProviderImpl$1.onSuccess(WebimSessionImpl.java:410)
        at com.webimapp.android.sdk.impl.WebimSessionImpl$RemoteHistoryProviderImpl$1.onSuccess(WebimSessionImpl.java:403)
        at com.webimapp.android.sdk.impl.backend.WebimActionsImpl$14.runCallback(WebimActionsImpl.java:405)
        at com.webimapp.android.sdk.impl.backend.WebimActionsImpl$14.runCallback(WebimActionsImpl.java:396)
        at com.webimapp.android.sdk.impl.backend.ActionRequestLoop$4.run(ActionRequestLoop.java:149)
        at com.webimapp.android.sdk.impl.WebimSessionImpl$ExecIfNotDestroyedHandlerExecutor$1.run(WebimSessionImpl.java:1349)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.os.HandlerThread.run(HandlerThread.java:65)
```

**Comment by djkovrik** [comment]
![image](https://user-images.githubusercontent.com/2476697/63928279-fa312100-ca57-11e9-8895-104caeb54dfd.png)

**Comment by webim** [comment]
Thank you for bug report.
We are working on this issue.

**Comment by webim** [comment]
Bug fixed in version 3.31.2.
